src/snowflake/cli/_plugins/nativeapp/entities/application_package.py

```diff
@@ -266,7 +272,7 @@ def action_validate(
             post_deploy_hooks=model.meta and model.meta.post_deploy,
             package_scripts=[],  # Package scripts are not supported in PDFv2
             policy=policy,
-            use_scratch_stage=True,
+            use_scratch_stage=use_scratch_stage,
```

Comment:
had to make this a param since we want `False` in tests

Reply:
why is that? Didn't get it from the diff.

Reply:
in tests we don't want to call `deploy()` on the package, so we need to be able to pass `False` here. In the old `NativeAppManager.validate()`, the `use_scratch_stage` parameter defaults to `False`: https://github.com/snowflakedb/snowflake-cli/blob/b516dc29e59c8615d9e7497c9054e7de20b0aec2/src/snowflake/cli/_plugins/nativeapp/manager.py#L321

Reply:
this doesn't change prod code since the parameter now defaults to `True` in v2